visvis2/material/_triangle.py

```diff
@@ -21,16 +34,34 @@ def vertex_shader(
 def fragment_shader(
     in_color: (RES_INPUT, 0, "vec3"),
     out_color: (RES_OUTPUT, 0, "vec4"),
-    # u_color: (RES_UNIFORM, 0, "vec3"),
+    uniforms: (RES_UNIFORM, (1, 0), uniform_type),
 ):
-    out_color = vec4(in_color, 0.1)  # noqa
+    color = uniforms.color
+    out_color = vec4(color, 0.1)  # noqa
 
 
 class TriangleMaterial(Material):
     def __init__(self):
         super().__init__()
-        self.shaders = {
-            "vertex": vertex_shader,
-            "fragment": fragment_shader,
-        }
-        self.primitive_topology = wgpu.PrimitiveTopology.triangle_list
+
+        uniforms_array = array_from_shader_type(uniform_type)
+        self.uniforms = BufferWrapper(uniforms_array, mapped=1, usage="uniform")
+        self.set_color((1, 0, 0))
+
+    def set_color(self, color):
+        self.uniforms.data["color"] = color
+
+    def get_wgpu_info(self, obj):
+
+        n = 3
+
+        return [
+            {
+                "vertex_shader": vertex_shader,
+                "fragment_shader": fragment_shader,
+                "primitive_topology": wgpu.PrimitiveTopology.triangle_list,
+                "indices": range(n),
+                "bindings1": [self.uniforms],
+                "target": None,  # default
+            },
+        ]
```

Comment:
I guess this whole thing should be moved to the example, just like you moved the WorldObject subclass. it doesn't belong in the library

Reply:
I agree.